finalProExe.py

```diff
@@ -0,0 +1,52 @@
+import argparse
+import asyncio
+import json
+import openai
+from pptx import Presentation
+
+def extract_text_from_slide(slide):
+    return  " ".join([run.text for shape in slide.shapes if shape.has_text_frame for paragraph in shape.text_frame.paragraphs for run in paragraph.runs]).strip()
```

Comment:
3 levels of nesting are not recommended 
you can divide the function to multiple function

Reply:
I divided the project into modules and merged branches